<launch>
  <arg name="use_keyboard" default="false"/> <!-- use the keyboard ? -->
  <arg name="use_joystick" default="false"/> <!-- use the jostick ? -->

  <!-- Joystick device -->
  <arg name="joystick_dev" default="/dev/input/by-id/usb-ACRUX_HAMA_X-Style_Pad-joystick" doc="joystick device"/>
  <!--arg name="joystick_dev" default="/dev/input/by-id/usb-Nintendo_Co.__Ltd._Pro_Controller_000000000001-joystick" doc="joystick device"/-->  
  <!-- Joystick Mapping: "Hama" | "F710" | "Switch" -->
  <arg name="joystick_map" default="Hama" doc="joystick mapping"/>


  <!-- CozmoDriver -->
  <!--include file="$(find cozmo_driver)/launch/cozmo_driver.launch">
  </include-->
  <node pkg="cozmo" type="cozmo_driver.py" name="cozmo_driver" output="screen">
  </node>

  <node pkg="cozmo" type="cube_orb_detector.py" name="cube_orb_detector" output="screen">
    <remap from="/in" to="/camera"/>
  </node>
  <node pkg="cozmo" type="cube_blob_detector.py" name="cube_blob_detector" output="screen">
    <remap from="/in" to="/camera"/>
  </node>
  
  <!-- Cozmo joystick node ************************************************** -->
  <group if="$(arg use_joystick)">
    <node pkg="joy" type="joy_node" name="joy" output="screen">
      <param name="dev" value="$(arg joystick_dev)"/>
      <param name="autorepeat_rate" value="10"/><!-- 10Hz-->
    </node>
    <!-- Cozmo joystick node -->
      <node pkg="cozmo" type="cozmo_joy.py" name="cozmo_joy" output="screen">
      <param name="joymap" type="str" value="$(arg joystick_map)"/>
    </node>
  </group>

  <!-- Cozmo keyboard node ************************************************* -->
  <group if="$(arg use_keyboard)">
    <node pkg="cozmo" type="cozmo_key.py" name="cozmo_key" output="screen">
    </node>
  </group>

</launch>
